### PR TITLE
Add support for a prefix in Cython module targets

### PR DESCRIPTION
**Pull request by vyasr**
A package composed of subpackages may contain multiple Cython modules with the same filename at different paths i.e. in different parts of the package directory tree. Since we use the filenames to construct the CMake library targets, the resulting library target names will conflict unless there is a way to customize them. This PR adds that functionality. Note that because of the way that scikit-build's Cython is designed, we need to ensure that the generated shared library is named  according to the original filename; the prefix should only be applied to the CMake target to avoid conflicts within CMake.

### CODE REVIEW

File: testing/cython/create_modules/CMakeLists.txt [hunk 1]
```diff
@@ -38,3 +38,7 @@ rapids_cython_create_modules()
 rapids_cython_create_modules(
     SOURCE_FILES test.pyx
     )
+
+if(NOT TARGET test)
+  message(FATAL_ERROR "rapids_cython_init didn't create the prefixed target test")
```
Comment (robertmaynard) [inline testing/cython/create_modules/CMakeLists.txt:43]:
```suggestion
  message(FATAL_ERROR "rapids_cython_init didn't create the target `test`")
```

File: testing/cython/create_modules_with_prefix/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+rapids_cython_create_modules()
+
+# Test that we can use a prefix to generate a different library target.
+rapids_cython_create_modules(
+    SOURCE_FILES test.pyx
+    MODULE_PREFIX "test_"
+    )
+
+if(NOT TARGET test_test)
+  message(FATAL_ERROR "rapids_cython_init didn't create the prefixed library target test::test")
```
Comment (robertmaynard) [inline testing/cython/create_modules_with_prefix/CMakeLists.txt:44]:
```suggestion
  message(FATAL_ERROR "rapids_cython_init didn't create the prefixed library target `test_test`")
```

**Review by robertmaynard — COMMENTED** [review]
Overall looks great.

**Review by robertmaynard — APPROVED** [review]
LGTM, merge whenever you want

**Comment by vyasr** [comment]
Thanks @robertmaynard. I applied your suggestions and undid the changes to RAPIDS.cmake, so we should be good to merge unless you spot anything else on a final review.

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**